File: backend/VolunteerReport.Domain/Entities/ReportDetail.cs
namespace VolunteerReport.Domain.Entities;

public class ReportDetail: IBaseEntity
{
    public Guid Id { get; set; }
    public double Amount { get; set; }
    public decimal Cost { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? ModifiedAt { get; set; }
    public bool IsDeleted { get; set; }

    public Guid ReportId { get; set; }
    public Report Report { get; set; }
    public Guid CategoryId { get; set; }
    public ReportCategory Category { get; set; }
}

File: mobile/lib/models/report_detail.dart
import 'package:mobile/models/report_category.dart';

class ReportDetail {
  final int id;
  final int amount;
  final double cost;
  final ReportCategory reportCategory;

  ReportDetail({
    required this.id,
    required this.cost,
    required this.amount,
    required this.reportCategory,
  });
}
